docs/faq.md

```diff
@@ -1,21 +1,321 @@
+
+# Network
+
+## Can Fn do IO over HTTP or in general any TCP based communication?
+
+Like most functions platforms we support HTTP.
+
+## Can an Fn function connect to various endpoints to read/write data?
+
+Yes, a function is not restricted in what it can connect to.
+
+## Any limits on the network bandwidth?
+
+We currently don't yet offer Fn as a managed service which would manage network traffic.  Currently, there are no restrictions in the core Fn platform.
+
+# Performance
+
+## How long can a function run? Can we make it run forever?
+
+We support 'hot functions' (see the end of [this tutorial](https://github.com/fnproject/tutorials/blob/master/JavaFDKIntroduction/README.md) for an example).  Hot functions will continue to live if they are used but, if not, will eventually be cleaned up.
+
+Function timeout is configurable. Please see <https://github.com/fnproject/fn/blob/master/docs/function-file.md>
+
+## How does the service trace the liveliness of the function? If my function dies/crashes will the service provision it again?
+
+This is not a microservices platform so the notion is slightly different.  Functions are run when called.  If you call a function that doesn't exist then one will be started.
+
+## How does scaling work? Should this be initiated by a customer?
+
+Fn runs more instances of the function based on requests.  It is automatic.
+
+## When one host is overloaded, will this service automatically relocate the function to another host?
+
+When running on Kubernetes (K8s) workloads are spread across available hosts.
+
+Functions currently do not relocate. Please see our [FnLB project inside Fn Project](https://github.com/fnproject/fn/tree/master/fnlb) for how to route requests to available hosts. K8s can also help with this.
+
+# Java
+
+## Any Java runtime dependencies? Can we use concurrent libraries or any third party?
+
+We currently support Maven which allows you to specify your dependencies. Therefore, you can use concurrency and third party libraries.
+
+## What versions of Java are supported (JDK 8, JDK 9)?
+
+We default to Java 9 now but if you want Java 8, the runtime name is `java8` (when you init a function or in your `func.yaml`).
+
+# Persistence
+
+## Does your runtime provide any persistent store to save the function state information? 
+
+State management is not part of Fn but you can use any storage service or database to store data.
+
+## Can we build a state function which will update itself based on the previous computation? Will it persist across function restarts?
+
+The problem you face is the lack of a guarantee of which instance of a hot function is called.  Standard practice is to externalize state.  The State Service is essentially Oracle Coherence which offers excellent performance as you may know.
+
+Also check out Fn Flow if you’re writing Java (currently the primary supported language).
+
+## Do you provide any UI/tool to track our Fn status?
```

Comment:
This question doesn't belong in the Java section.

Reply:
Oops. Definitely missed this. Moved to General.